From 917b2b408fc78ae39b277e91362d9318dcf64ac8 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:40 -0400
Subject: [scsi] hpsa: fix bad endif placement in RAID 5 mapper code

Message-id: <1410186734-19986-50-git-send-email-thenzl@redhat.com>
Patchwork-id: 92064
O-Subject: [RHEL6.6 e-stor PATCH 49/83] hpsa: fix bad endif placement in RAID 5 mapper code
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

It caused the i/o request to always be counted as ineligible for
the accelerated i/o path on 32 bit systems and negatively affected
performance.

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 000ff7c25a15134286c8bedf3975fe79f09d1798)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 1b0db02..e194de6 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -3846,8 +3846,8 @@ static int hpsa_scsi_ioaccel_raid_map(struct ctlr_info *h,
 #else
   first_group = (first_block % stripesize) / r5or6_blocks_per_row;
   last_group = (last_block % stripesize) / r5or6_blocks_per_row;
-  if (first_group != last_group)
 #endif
+  if (first_group != last_group)
    return IO_ACCEL_INELIGIBLE;
 
   /* Verify request is in a single row of RAID 5/6 */
-- 
1.7.1